### PR TITLE
make --detect-commit the default behaviour

### LINKED ISSUE #19: make --detect-commit the default
this is usually the sensible behaviour. Instruct users what to do if this fails (e.g. how to upload to HEAD of the main branch instead)

### PR DESCRIPTION
**Pull request by karottenreibe**
Fixes #19 #22 

- [x] Changes are tested adequately
- [ ] Teamscale documentation updated in case of relevant user-visible changes
- [x] CHANGELOG.md updated
- [x] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566) not needed

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config or architecture file needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

**Comment by karottenreibe** [comment]
need to update TS docs after merging and releasing

File: src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java [hunk 1]
```diff
@@ -131,7 +132,7 @@ public void validate(ArgumentParser parser) throws ArgumentParserException {
             }
 
             if (hasMoreThanOneCommitOptionSet()) {
-                throw new ArgumentParserException("You used more than one of --detect-commit, --commit and --timestamp." +
+                throw new ArgumentParserException("You used more than one of --detect-commit, --commit and --branch-and-timestamp." +
```
Comment (saileru) [inline src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java:135]: As we removed the `detect-commit` option, we should not mention it here anymore, right?

**Review by saileru — CHANGES_REQUESTED** [review]
Nice 🙂 Only one minor comment.

**Comment by karottenreibe** [comment]
thanks @saileru fixed that :-)

**Pull request merged**